src/mds/Server.cc

```diff
@@ -2647,9 +2647,14 @@ void Server::dispatch_client_request(const MDRequestRef& mdr)
       auto it = mdr->batch_op_map->find(mask);
       auto new_batch_head = it->second->find_new_head();
       if (!new_batch_head) {
-	mdr->batch_op_map->erase(it);
-	return;
+        mdr->batch_op_map->erase(it);
+        dout(10) << __func__ << ": mask '" << mask
+                 << "' batch head is dead and there is no follower" << dendl;
+        return;
       }
+      dout(10) << __func__ << ": mask '" << mask
+               << "' batch head is dead and queue a new one "
+               << *new_batch_head << dendl;
       mds->finisher->queue(new C_MDS_RetryRequest(mdcache, new_batch_head));
```

Comment:
im wondering how did it queue a `nullptr` ref and nothing broke? is there any mechanism that discards null refs? or null refs(w.r.t to client reqs) is not an issue at all?

Reply:
Sorry I didn't get u here. It won't queue a `nullptr` ref here. If the `new_batch_head == nullptr` it will just erase it from the `batch_op_map` and `return`.